Makefile

```diff
@@ -70,6 +70,10 @@ ALL_SRC         := $(shell find . -name "*.go")
 ALL_SRC         += go.mod
 ALL_SCRIPTS     := $(shell find . -name "*.sh")
 
+MAIN_BRANCH	   = master
+MERGE_BASE     ?= $(shell git merge-base $(MAIN_BRANCH) HEAD)
+MODIFIED_FILES := $(shell git diff --name-status $(MERGE_BASE) -- | cut -f2)
```

Comment:
I don't think this is necessary. I already tested that this will work for linter issues in any commit in the PR: https://github.com/temporalio/temporal/pull/3901

Reply:
This is necessary to work locally and on CI.

Reply:
I see. In that case, it would be simpler to use `--new-from-rev=master`. We shouldn't have to write this logic ourselves since this is a very common usage pattern for golangci-lint.